### PR TITLE
test and fix tkinter initialization path

### LINKED ISSUE #80: Path in TkApp is wrong
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The pillow-feedstock is failing tests. @isuruf [noted](https://github.com/conda-forge/pillow-feedstock/pull/116#issuecomment-1094367845) that the path mangling is wrong.
> With the new file layout, https://foss.heptapod.net/pypy/pypy/-/blob/branch/py3.8/lib_pypy/_tkinter/app.py#L47 seems to need an extra dirname

Not only that, the subdirectories are `tcl8.6` and `tk8.6`, not the bare `tcl` and `tk` . I wonder how CPython deals with this?

### Installed packages

```shell
conda create -n pypy38 pypy python=3.8
```

### Environment info

```shell
$ conda info

     active environment : pypy38
    active env location : /home/matti/miniconda3/envs/pypy38
            shell level : 2
       user config file : /home/matti/.condarc
 populated config files : /home/matti/.condarc
          conda version : 4.10.3
    conda-build version : not installed
         python version : 3.9.7.final.0
       virtual packages : __cuda=11.2=0
                          __linux=5.4.0=0
                          __glibc=2.31=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/matti/miniconda3  (writable)
      conda av data dir : /home/matti/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/pytorch/linux-64
                          https://conda.anaconda.org/pytorch/noarch
          package cache : /home/matti/miniconda3/pkgs
                          /home/matti/.conda/pkgs
       envs directories : /home/matti/miniconda3/envs
                          /home/matti/.conda/envs
               platform : linux-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.9.7 Linux/5.4.0-107-generic ubuntu/20.04.4 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #80 and adds a test to make sure tkinter works.

### CODE REVIEW

**Comment by mattip** [comment]
@conda-forge-admin, please rerender

**Comment by mattip** [comment]
Builds on linux are [failing to run `'import tkinter; tkinter.Tk()'`](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=490097&view=logs&j=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de&t=10fc5aa2-324e-5982-4c88-6b31fcab16b3&l=5097) because they are missing `xorg-libx11`. If that is a requirement for `tk`, why doesn't installing `tk` install `xorg-libx11`?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -91,6 +93,7 @@ test:
   commands:
     - pypy3 --help
     - pypy3 -c "import platform; print(platform._sys_version())"
+    - pypy3 -c "import tkinter; tkinter.Tk()"
```
Comment (mattip) [inline recipe/meta.yaml:96]: Remove this to allow linux builds to pass

File: recipe/meta.yaml [hunk 2]
```diff
@@ -77,6 +78,7 @@ requirements:
     - openssl
     - sqlite
     - tk
+    - xorg-libx11  # [linux]
```
Comment (isuruf) [inline recipe/meta.yaml:81]: Remove this?

File: recipe/meta.yaml (hunk 1 above)
Comment (mattip) [reply recipe/meta.yaml]: done

**Comment by mattip** [comment]
This seems to be working, and on macOS and windows now tests `tkinter.Tk()`. It removed some extra work done to build portable PyPy, (which is not needed on conda), and properly links to the tk/tcl libs on macOS.

Anyone wish to review?

**Pull request merged**